hazelcast/src/test/java/com/hazelcast/cache/impl/journal/EventJournalCacheDataStructureAdapter.java

```diff
@@ -0,0 +1,109 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.cache.impl.journal;
+
+import com.hazelcast.cache.HazelcastExpiryPolicy;
+import com.hazelcast.cache.ICache;
+import com.hazelcast.cache.impl.CacheService;
+import com.hazelcast.cache.journal.EventJournalCacheEvent;
+import com.hazelcast.core.ICompletableFuture;
+import com.hazelcast.internal.journal.EventJournalInitialSubscriberState;
+import com.hazelcast.internal.journal.EventJournalReader;
+import com.hazelcast.journal.EventJournalDataStructureAdapter;
+import com.hazelcast.projection.Projection;
+import com.hazelcast.ringbuffer.ReadResultSet;
+import com.hazelcast.spi.ObjectNamespace;
+
+import javax.cache.Cache;
+import java.util.AbstractMap.SimpleImmutableEntry;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+public class EventJournalCacheDataStructureAdapter<K, V>
+        implements EventJournalDataStructureAdapter<K, V, EventJournalCacheEvent<K, V>> {
+
+    private final ICache<K, V> cache;
+
+    public EventJournalCacheDataStructureAdapter(ICache<K, V> cache) {
+        this.cache = cache;
+    }
+
+    @Override
+    public int size() {
+        return cache.size();
+    }
+
+    @Override
+    public V get(K key) {
+        return cache.get(key);
+    }
+
+    @Override
+    public V put(K key, V value) {
+        return cache.getAndPut(key, value);
+    }
+
+    @Override
+    public V put(K key, V value, long ttl, TimeUnit timeunit) {
+        return cache.getAndPut(key, value, new HazelcastExpiryPolicy(ttl, ttl, ttl, timeunit));
+    }
+
+    @Override
+    public ObjectNamespace getNamespace() {
+        return CacheService.getObjectNamespace(cache.getPrefixedName());
+    }
+
+    @Override
+    public Set<Entry<K, V>> entrySet() {
+        return getEntries(cache);
+    }
+
+    @Override
+    public V remove(K key) {
+        return cache.getAndRemove(key);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public ICompletableFuture<EventJournalInitialSubscriberState> subscribeToEventJournal(int partitionId) {
+        return ((EventJournalReader<?>) cache).subscribeToEventJournal(partitionId);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public <T> ICompletableFuture<ReadResultSet<T>> readFromEventJournal(
+            long startSequence, int minSize, int maxSize, int partitionId,
+            com.hazelcast.util.function.Predicate<? super EventJournalCacheEvent<K, V>> predicate,
+            Projection<? super EventJournalCacheEvent<K, V>, ? extends T> projection) {
+        return ((EventJournalReader<EventJournalCacheEvent<K, V>>) cache)
```

Comment:
My glasses just cracked :)

Could you do a argument per method? Then at least it is clear how many arguments and what their position is.

Reply:
I haven't split it into multiple methods but I've tried to make it look nicer. Hopefully it works. I can split it into multiple methods if necessary.